### DISCUSSION TITLE
How to implement createWeb3Modal() function in Nextjs 13

### DISCUSSION THREAD

**Discussion by abdullahmehboob20s** [WalletConnect/walletconnect-monorepo#3786] [Developer Support] [answered]
I am using Nextjs 13 and set up everything written in [docs](https://docs.walletconnect.com/web3modal/react/about). But i am stuck on this [point](https://docs.walletconnect.com/web3modal/react/about#implementation) where I am having to put `createWeb3Modal({ wagmiConfig, projectId, chains })` in my layout.js file. I am receiving this error again and again. And this is the only I am getting. Please help me fix it.

![image](https://github.com/WalletConnect/walletconnect-monorepo/assets/64467248/59bccc38-75e3-490d-8fef-9062c0fd39cb)

**Comment by glitch-txs** [comment, marked as answer]
'use client' directive.
